File: booklab-backend/src/test/java/nl/tudelft/booklab/backend/ocr/tesseract/BasicTesseractExampleTest.java
package nl.tudelft.booklab.backend.ocr.tesseract;

import kotlin.io.ByteStreamsKt;
import kotlin.io.ConstantsKt;
import org.bytedeco.javacpp.*;
import org.junit.Test;

import java.io.InputStream;
import java.net.URL;
import java.nio.ByteBuffer;

import static org.bytedeco.javacpp.lept.*;
import static org.bytedeco.javacpp.tesseract.*;
import static org.junit.Assert.assertTrue;

public class BasicTesseractExampleTest {

    @Test
    public void givenTessBaseApi_whenImageOcrd_thenTextDisplayed() throws Exception {
        BytePointer outText;

        TessBaseAPI api = new TessBaseAPI();
        // Initialize tesseract-ocr with English, without specifying tessdata path
        if (api.Init(".", "ENG") != 0) {
            System.err.println("Could not initialize tesseract.");
            System.exit(1);
        }

        // Open input image with leptonica library
        URL path = BasicTesseractExampleTest.class.getResource("/test.png");
        String pathstring = path.getPath();
        pathstring = pathstring.substring(1);
        PIX image = pixRead(pathstring);

        api.SetImage(image);
        // Get OCR result
        outText = api.GetUTF8Text();
        String string = outText.getString();
        assertTrue(!string.isEmpty());
        System.out.println("OCR output:\n" + string);

        // Destroy used object and release memory
        api.End();
        outText.deallocate();
        pixDestroy(image);
    }
}

File: booklab-backend-detection/src/test/kotlin/nl/tudelft/booklab/backend/ocr/tesseract/BasicTesseractExampleTest.kt
/*
 * Copyright 2018 The BookLab Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package nl.tudelft.booklab.backend.ocr.tesseract;

import org.bytedeco.javacpp.lept.pixDestroy
import org.bytedeco.javacpp.lept.pixRead
import org.bytedeco.javacpp.tesseract
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Test

internal class BasicTesseractExampleTest {
    @Test
    fun `given TessBaseApi when image OCRd then text displayed`() {
        val api = tesseract.TessBaseAPI()
        // Initialize tesseract-ocr with English, without specifying tessdata path
        if (api.Init(".", "ENG") != 0) {
            throw IllegalStateException("Failed to initialise Tesseract")
        }

        // Open input image with leptonica library
        val path = BasicTesseractExampleTest::class.java.getResource("/tesseract.png");
        val pathString = path.getPath().substring(1);
        val image = pixRead(pathString);

        api.SetImage(image);
        // Get OCR result
        val outText = api.GetUTF8Text();
        val string = outText.getString();
        assertFalse(string.isEmpty());

        println("OCR output:")
        println(string)

        // Destroy used object and release memory
        api.End();
        outText.deallocate();
        pixDestroy(image);
    }
}
